src/formatter/formatCommaPositions.ts

```diff
@@ -68,7 +68,11 @@ function formatTabular(commaLines: string[]): string[] {
     if (i === commaLines.length - 1) {
       return line; // do not add comma for last item
     }
-    return line + ' '.repeat(maxLineLength - line.length - 1) + ',';
+    const commentMatch = /\s*--/.exec(line);
+    const endOfLinePosition = commentMatch ? commentMatch.index : line.length;
+    return `${line.slice(0, endOfLinePosition)}${' '.repeat(
+      maxLineLength - line.length - 1
+    )},${line.slice(endOfLinePosition + 1)}`;
```

Comment:
I have trouble understanding this code. It attempts to do something with comments in the `commaPosition:tabular` case. But there are no tests that cover this scenario.

I tried to write a test for this, but it produced a pretty strange result:

```sql
SELECT
  alpha         ,
  beta,--comment
  delta         ,
  epsilon       ,
  iota
```

Reply:
>  have trouble understanding this code.
This part was refactored to make it easier to undestand

>  I tried to write a test for this, but it produced a pretty strange result:
This case was added to tests. Also I changed code to make it work as expected.

If you'll find other use cases, I'll add them to tests with corresponding fixes.